### PR TITLE
Argon2 rewrite

### PR DESCRIPTION
**Pull request by OhGodAPet**
Currently REQUIRES AVX2.

### CODE REVIEW

**Comment by spencerlievens** [comment]
utACK

**Comment by plaxton** [comment]
utACK

**Comment by spencerlievens** [comment]
utACK

File: src/crypto/argon2d/argon2.c [hunk 1]
```diff
@@ -377,3 +377,326 @@ size_t argon2_encodedlen(uint32_t t_cost, uint32_t m_cost, uint32_t parallelism,
          numlen(t_cost) + numlen(m_cost) + numlen(parallelism) +
          b64len(saltlen) + b64len(hashlen);
 }
+
+
+///////////////////////////
```
Comment (spencerlievens) [inline src/crypto/argon2d/argon2.c]: utACK

**Pull request merged**